### PR TITLE
Update to Vue 3

### PR DESCRIPTION
**Pull request by rictic**

### CODE REVIEW

File: libraries/vue/package.json [hunk 1]
```diff
@@ -14,11 +14,11 @@
     "karma-firefox-launcher": "2.1.2",
     "karma-sourcemap-loader": "0.3.8",
     "karma-webpack": "5.0.0",
-    "vue-template-compiler": "2.6.14",
+    "@vue/compiler-sfc": "^3.1.0",
```
Comment (AndrewJakubowicz) [inline libraries/vue/package.json:17]: Nit: Is this dependency required?

Ref: https://www.npmjs.com/package/@vue/compiler-sfc#:~:text=vue/compiler%2Dsfc%20deep%20import

File: libraries/vue/src/basic-tests.js [hunk 2]
```diff
@@ -70,60 +76,72 @@ describe("basic support", function() {
...
-      let vm = new ComponentWithChildrenRerender().$mount(scratch);
-      let root = vm.$el;
-      let wc = root.querySelector("#wc");
-      await vm.$nextTick();
+      const app = createApp(ComponentWithChildrenRerender)
+      app.config.compilerOptions.isCustomElement = isCustomElement;
+      app.mount(scratch);
+      let wc = scratch.querySelector("#wc");
+      await nextTick();
+      await nextTick();
```
Comment (AndrewJakubowicz) [inline libraries/vue/src/basic-tests.js:93]: Nit: Need tick twice?

File: libraries/vue/src/advanced-tests.js [hunk 3]
```diff
@@ -68,61 +76,66 @@ describe("advanced support", function() {
...
     it("can declaratively listen to a CAPScase DOM event dispatched by a Custom Element", async function() {
       this.weight = 1;
-      let vm = new ComponentWithDeclarativeEvent().$mount(scratch);
-      let root = vm.$el;
-      let wc = root.querySelector("#wc");
-      let handled = root.querySelector("#caps");
+      const app = createApp(ComponentWithDeclarativeEvent)
+      app.config.compilerOptions.isCustomElement = isCustomElement;
+      app.mount(scratch);
+      let wc = scratch.querySelector("#wc");
```
Comment (AndrewJakubowicz) [inline libraries/vue/src/advanced-tests.js:121]: Nits that don't need action. Prefer `const` to `let` here and on handled.

File: libraries/vue/package.json [hunk 4]
```diff
@@ -14,11 +14,11 @@
...
     "webpack": "5.65.0"
   },
   "dependencies": {
     "@webcomponents/webcomponentsjs": "2.6.0",
-    "vue": "2.6.14"
+    "vue": "^3.1.0"
```
Comment (AndrewJakubowicz) [inline libraries/vue/package.json:22]: Nit: Bump minor version to 3.2.30

**Review by AndrewJakubowicz — APPROVED** [review]
A couple minor nits. Thank you for doing this!

File: libraries/vue/src/basic-tests.js (hunk 2 above)
Comment (rictic) [reply libraries/vue/src/basic-tests.js:94]: Yep, needed. Will leave a comment.

File: libraries/vue/package.json (hunk 4 above)
Comment (rictic) [reply libraries/vue/package.json:21]: the '^' operator takes care of that, and the package-lock.json actually specifies the version that we test with

**Pull request merged**